code/main.py

```diff
@@ -158,159 +147,231 @@ def main():
+                )
+                test_module = importlib.util.module_from_spec(spec=test_spec)
+                test_spec.loader.exec_module(test_module)
+                test_function = getattr(test_module, test_file_function_name, None)
+            except ModuleNotFoundError as exception:
+                print(f"::error::Could not load python script in your repository which defines theweb service tests (Script: /{test_file_path}, Function: {test_file_function_name}()): {exception}")
+                raise AMLConfigurationException(f"Could not load python script in your repository which defines the web service tests (Script: /{test_file_path}, Function: {test_file_function_name}()): {exception}")
+            except FileNotFoundError as exception:
+                print(f"::error::Could not load python script or function in your repository which defines the web service tests (Script: /{test_file_path}, Function: {test_file_function_name}()): {exception}")
+                raise AMLConfigurationException(f"Could not load python script or function in your repository which defines the web service tests (Script: /{test_file_path}, Function: {test_file_function_name}()): {exception}")
+            except AttributeError as exception:
+                print(f"::error::Could not load python script or function in your repository which defines the web service tests (Script: /{test_file_path}, Function: {test_file_function_name}()): {exception}")
+                raise AMLConfigurationException(f"Could not load python script or function in your repository which defines the web service tests (Script: /{test_file_path}, Function: {test_file_function_name}()): {exception}")
+
+            # Load experiment config
+            print("::debug::Loading experiment config")
+            try:
+                test_function(service)
+            except TypeError as exception:
+                print(f"::error::Could not load experiment config from your module (Script: /{test_file_path}, Function: {test_file_function_name}()): {exception}")
+                raise AMLConfigurationException(f"Could not load experiment config from your module (Script: /{test_file_path}, Function: {test_file_function_name}()): {exception}")
+            except Exception as exception:
+                print(f"::error::The webservice tests did not complete successfully: {exception}")
+                raise AMLDeploymentException(f"The webservice tests did not complete successfully: {exception}")
+
+        # Deleting service if desired
+        if parameters.get("delete_service_after_deployment", False):
+            service.delete()
+        else:
+            # Creating outputs
+            print("::debug::Creating outputs")
+            print(f"::set-output name=service_scoring_uri::{service.scoring_uri}")
+            print(f"::set-output name=service_swagger_uri::{service.swagger_uri}")
+
+    # Pulling Docker image
+    if parameters.get("create_image", None) is not None:
```

Comment:
also need to check if it is not false?

Reply:
if user deployment happens then still the image is created, do we also want to provide the acr information in case user has asked for deployment only?

Reply:
No probably not, right?
What's makes you think that this is required? I am curious to find out whether this is required.

Reply:
my bad.. i thought it is a boolean var! .you are write it is not required.